File: backend/examserver/src/main/java/com/exam/service/StatisticsService.java
package com.exam.service;

public interface StatisticsService {

    // Get total marks by category
    Integer getTotalMarksByCategory(Long categoryId);

    // Get total number of attempts for the category
    Long getTotalAttemptsByCategory(Long categoryId);

    // Get average score by category
    Double getAverageScoreByCategory(Long categoryId);

    // Get total number of correct answers in the category
    Integer getTotalCorrectAnswersByCategory(Long categoryId);

    // Get total number of questions attempted in the category
    Integer getTotalQuestionsAttemptedByCategory(Long categoryId);
}

File: frontend/src/app/services/statistics.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class StatisticsService {


  constructor(private http: HttpClient) {}

  getTotalMarks(categoryId: number): Observable<number> {
    return this.http.get<number>(`${baseUrl}/statistics/total-marks/${categoryId}`);
  }

  getTotalAttempts(categoryId: number): Observable<number> {
    return this.http.get<number>(`${baseUrl}/statistics/total-attempts/${categoryId}`);
  }

  getAverageScore(categoryId: number): Observable<number> {
    return this.http.get<number>(`${baseUrl}/statistics/average-score/${categoryId}`);
  }

  getTotalCorrectAnswers(categoryId: number): Observable<number> {
    return this.http.get<number>(`${baseUrl}/statistics/total-correct-answers/${categoryId}`);
  }

  getTotalQuestionsAttempted(categoryId: number): Observable<number> {
    return this.http.get<number>(`${baseUrl}/statistics/total-questions-attempted/${categoryId}`);
  }
}
